lychee-lib/src/collector.rs

```diff
@@ -222,11 +222,9 @@ pub async fn collect_links(
     skip_missing_inputs: bool,
     max_concurrency: usize,
 ) -> Result<HashSet<Request>> {
-    let base_url = if let Some(url) = base_url {
-        Some(Url::parse(&url).map_err(|e| (url, e))?)
-    } else {
-        None
-    };
+    let base_url = base_url
```

Comment:
Again, not necessarily more readable.

Reply:
I disagree but again, I guess it's a matter of opinion if you think the functional or the imperative style is more readable